src/m365/context/commands/context-init.spec.ts

```diff
@@ -0,0 +1,135 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../appInsights';
+import auth from '../../../Auth';
+import { Logger } from '../../../cli/Logger';
+import Command, { CommandError } from '../../../Command';
+import { pid } from '../../../utils/pid';
+import { sinonUtil } from '../../../utils/sinonUtil';
+import * as fs from 'fs';
+import commands from '../commands';
+const command: Command = require('./context-init');
+
+describe(commands.INIT, () => {
+  let log: any[];
+  let logger: Logger;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName,
+      fs.existsSync,
+      fs.readFileSync,
+      fs.writeFileSync
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.INIT), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it(`saves context info in the .m365rc.json file in the current folder when requested. Creates the file if it doesn't exist`, async () => {
```

Comment:
Let's shorten the test description and avoid repeating the first sentence which will make it harder for us to find the right test